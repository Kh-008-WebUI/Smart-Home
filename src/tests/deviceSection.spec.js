import React from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import DevicesSection from '../components/DevisesSection/DevisesSection';
import configureStore from 'redux-mock-store';
import {createStore} from 'redux';

const state = {
  searchAndFilter: {
    filterOption: 'all',
    searchValue: ''
  },
  devicesList: {
    devices: [
      {
        _id : '599c8094258df03990fae071',
        name : 'Device Test',
        location : 'Parking',
        status : false,
        createdDate : 'August 22, 2017',
        createdBy : 'Bruce',
        items : [
          {
            description : 'Toggle',
            data : false,
            name : 'Toggle'
          },
          {
            description : 'Value',
            data : 'hello there',
            name : 'Value'
          },
          {
            description : 50,
            params : {
              maxValue : 100,
              minValue : 0
            },
            data : 50,
            name : 'Range'
          }
        ],
        views : 1
      },
      {
        _id : '589c8094258df03990fae071',
        name : 'Device Test 2',
        location : 'Parking',
        status : true,
        createdDate : 'August 22, 2017',
        createdBy : 'Bruce',
        items : [
          {
            description : 'Toggle',
            data : false,
            name : 'Toggle'
          },
          {
            description : 'Value',
            data : 'hello there',
            name : 'Value'
          },
          {
            description : 50,
            params : {
              maxValue : 100,
              minValue : 0
            },
            data : 50,
            name : 'Range'
          }
        ],
        views : 1
      },
      {
        _id : '589c8094258df03990fae071',
        name : 'Device Test 3',
        location : 'Parking',
        status : false,
        createdDate : 'August 22, 2017',
        createdBy : 'Bruce',
        items : [
          {
            description : 'Toggle',
            data : false,
            name : 'Toggle'
          },
          {
            description : 'Value',
            data : 'hello there',
            name : 'Value'
          },
          {
            description : 50,
            params : {
              maxValue : 100,
              minValue : 0
            },
            data : 50,
            name : 'Range'
          }
        ],
        views : 1
      },
      {
        _id : '589c8094258df03990fae071',
        name : 'Device Test 4',
        location : 'Parking',
        status : false,
        createdDate : 'August 22, 2017',
        createdBy : 'Bruce',
        items : [
          {
            description : 'Toggle',
            data : false,
            name : 'Toggle'
          },
          {
            description : 'Value',
            data : 'hello there',
            name : 'Value'
          },
          {
            description : 50,
            params : {
              maxValue : 100,
              minValue : 0
            },
            data : 50,
            name : 'Range'
          }
        ],
        views : 1
      },
      {
        _id : '589c8094258df03990fae073',
        name : 'Device Test 5',
        location : 'Parking',
        status : true,
        createdDate : 'August 22, 2017',
        createdBy : 'Bruce',
        items : [
          {
            description : 'Toggle',
            data : false,
            name : 'Toggle'
          },
          {
            description : 'Value',
            data : 'hello there',
            name : 'Value'
          },
          {
            description : 50,
            params : {
              maxValue : 100,
              minValue : 0
            },
            data : 50,
            name : 'Range'
          }
        ],
        views : 1
      }
    ]
  }
};
const mockStore = configureStore()
let store,container

describe('DevicesSection Snapshot', () => {
  beforeEach(()=>{
    store = mockStore(state);
    container = shallow(<DevicesSection store={store} />)
  })

  it ('capturing Snapshot of DevicesSection', () => {
    expect(container).toMatchSnapshot();
  })
});