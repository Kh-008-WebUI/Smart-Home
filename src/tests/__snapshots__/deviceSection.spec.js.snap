// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevicesSection Snapshot capturing Snapshot of DevicesSection 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="device-list"
>
    <h1
        className="device-list__title"
    >
        Your devices
    </h1>
    <Connect(ListHeader)
        history={undefined}
        location={undefined}
        match={undefined}
        quantity={undefined}
    />
    <section
        className="device-list__content"
    />
    <Popup
        header="Confirm the action"
        popupShown={undefined}
        setPopupShown={undefined}
        text="Are you sure you want to remove the device?"
    >
        <Button
            className="btn popup__btn"
            innerText="Ok"
            okHandler={[Function]}
            setPopupShown={undefined}
        />
        <Button
            className="btn btn--default popup__btn"
            innerText="Cancel"
            okHandler={[Function]}
        />
    </Popup>
    <Message
        clearStatus={undefined}
        header="Error"
    />
</section>,
  "nodes": Array [
    <section
      className="device-list"
>
      <h1
            className="device-list__title"
      >
            Your devices
      </h1>
      <Connect(ListHeader)
            history={undefined}
            location={undefined}
            match={undefined}
            quantity={undefined}
      />
      <section
            className="device-list__content"
      />
      <Popup
            header="Confirm the action"
            popupShown={undefined}
            setPopupShown={undefined}
            text="Are you sure you want to remove the device?"
      >
            <Button
                  className="btn popup__btn"
                  innerText="Ok"
                  okHandler={[Function]}
                  setPopupShown={undefined}
            />
            <Button
                  className="btn btn--default popup__btn"
                  innerText="Cancel"
                  okHandler={[Function]}
            />
      </Popup>
      <Message
            clearStatus={undefined}
            header="Error"
      />
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DevicesSection
        store={
                Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DevicesSection {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="device-list"
>
          <h1
                    className="device-list__title"
          >
                    Your devices
          </h1>
          <Connect(ListHeader)
                    history={undefined}
                    location={undefined}
                    match={undefined}
                    quantity={undefined}
          />
          <section
                    className="device-list__content"
          />
          <Popup
                    header="Confirm the action"
                    popupShown={undefined}
                    setPopupShown={undefined}
                    text="Are you sure you want to remove the device?"
          >
                    <Button
                              className="btn popup__btn"
                              innerText="Ok"
                              okHandler={[Function]}
                              setPopupShown={undefined}
                    />
                    <Button
                              className="btn btn--default popup__btn"
                              innerText="Cancel"
                              okHandler={[Function]}
                    />
          </Popup>
          <Message
                    clearStatus={undefined}
                    header="Error"
          />
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="device-list"
>
          <h1
                    className="device-list__title"
          >
                    Your devices
          </h1>
          <Connect(ListHeader)
                    history={undefined}
                    location={undefined}
                    match={undefined}
                    quantity={undefined}
          />
          <section
                    className="device-list__content"
          />
          <Popup
                    header="Confirm the action"
                    popupShown={undefined}
                    setPopupShown={undefined}
                    text="Are you sure you want to remove the device?"
          >
                    <Button
                              className="btn popup__btn"
                              innerText="Ok"
                              okHandler={[Function]}
                              setPopupShown={undefined}
                    />
                    <Button
                              className="btn btn--default popup__btn"
                              innerText="Cancel"
                              okHandler={[Function]}
                    />
          </Popup>
          <Message
                    clearStatus={undefined}
                    header="Error"
          />
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DevicesSection
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
}
`;
